frontend/src/components/RegisterForm.tsx

```diff
@@ -0,0 +1,254 @@
+import React, { useState } from 'react'
+import * as Yup from 'yup'
+import { Formik, Form, FormikProps } from 'formik'
+import {
+  Grid,
+  TextField,
+  Button,
+  makeStyles,
+  createStyles,
+} from '@material-ui/core'
+import { useMutation } from '@apollo/client'
+import { REGISTER } from '../graphql/mutations'
+
+const stylesInUse = makeStyles(() =>
+  createStyles({
+    root: {
+      maxWidth: '500px',
+      color: 'white',
+      display: 'block',
+      margin: '0 auto',
+    },
+    textField: {
+      '& > *': {
+        width: '100%',
+      },
+    },
+    registerButton: {
+      marginTop: '30px',
+    },
+    title: { textAlign: 'center' },
+    successMessage: { color: 'green' },
+    errorMessage: { color: 'red' },
+  })
+)
+
+interface MyFormStatus {
+  type: string
+  message: string
+}
+
+interface MyFormStatusProps {
+  [key: string]: MyFormStatus
+}
+
+interface MyRegisterForm {
+  username: string
+  email: string
+  password: string
+  confirmPassword: string
+}
+
+const formStatusProps: MyFormStatusProps = {
+  error: {
+    message: 'Registeration failed. Please try again.',
+    type: 'error',
+  },
+  success: {
+    message: 'Registered successfully.',
+    type: 'success',
+  },
+}
+
+const RegisterForm = () => {
+  const [formStatus, setFormStatus] = useState<MyFormStatus>({
+    message: '',
+    type: '',
+  })
+  const classes = stylesInUse()
+  const [showFormStatus, setShowFormStatus] = useState(false)
+
+  const [
+    registerUser,
+    { loading: registerLoading, data: registerData },
+  ] = useMutation(REGISTER)
+
+  const createNewAccount = async (
+    data: MyRegisterForm,
+    resetForm: Function
+  ) => {
+    try {
+      await registerUser({
+        variables: {
+          username: data.username,
+          email: data.email,
+          password: data.password,
+        },
+      })
+    } catch (error) {
+      setFormStatus(formStatusProps.error)
+    }
+    console.log('register loading', registerLoading)
+    console.log('register data', registerData)
+    setFormStatus(formStatusProps.success)
+    resetForm({})
+    setShowFormStatus(true)
+  }
+
+  const UserSchema = Yup.object().shape({
```

Comment:
Should we move Yup schemas to different file to separate logic from react component?

Reply:
I refactored Yup schema to different file.